### ISSUE TITLE
Cant play a playlist

### ISSUE THREAD

**Issue by robotrono** [Splamy/TS3AudioBot#31]
Well the bot works fine, but when I try to play a playlist I get this error:

```
[00:09:23]  Debug: MB Got message from Akira: !list play
[00:09:24]Warning: TH Web error: [403] Forbidden
```
Edit: It's a problem with Youtube-DL? 

```
# youtube-dl -U
youtube-dl is up-to-date (2016.08.01)
```

**Comment by Splamy** [comment]
The first warning is not a problem, it just indicates that my script couldn't access youtube and ytdl will try to.
Should not be a problem with ytdl, do you have any songs in the playlist ?  
What do you get when you write `!list show`

**Comment by robotrono** [comment]
Yes, this is my playlist https://www.youtube.com/playlist?list=PLbbZzFSu3a-i-xc5VARuamD-Ds11-KXTm

```
<17:18:00> "Akira": !list show
<17:18:00> "TS3AudioBot": Playlist: "PLbbZzFSu3a-i-xc5VARuamD-Ds11-KXTm" with 387 songs.
0: twenty one pilots: Ride (Video)
1: High Performance - One More Time
2: Takako.MatsuーAshita haru ga kitara
3: Hanayamata opening  song full
4: NOMA - Brain Power
5: EDEN - drugs
6: DAOKO 『BANG!』 Music Video［HD］
7: Offramp - Tataki
8: Noisestorm - This Feeling
9: Yunomi - Oedo Controller ft. TORIENA
```

**Comment by Splamy** [comment]
Well I can tell you the typical "works fine for me" :/
Does actually anything come _after _ the error? It should look something like this:
```
[04:35:44]Warning: TH Web error: [403] Forbidden
[04:35:44]  Debug: YT Ruined!
[04:35:45]  Debug: YT Saved!
```
If not, make sure you have the latest dev commit and rebuilt it with xbuild. (And comment out the two webinterface lines if the other problem still persists)

**Comment by robotrono** [comment]
I get this error:

```
<17:09:53> "Akira": !play https://www.youtube.com/watch?v=Pw-0pbY9JeU
<17:09:53> "TS3AudioBot": Could not load (/usr/bin/python: can't open file '/usr/local/bin/youtube-dl/youtube_dl/__main__.py': [Errno 20] Not a directory
)
```

But I followed this download:

https://rg3.github.io/youtube-dl/download.html

```
sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl

sudo chmod a+rx /usr/local/bin/youtube-dl
```

**Comment by Splamy** [comment]
Can you look in your config file, one line should look like that:
`YoutubeFactory::youtubedlpath=/usr/local/bin/youtube-dl`

does the bot and ytdl belong to the same user ?

**Comment by robotrono** [comment]
This is my config file:

```
AudioFramework::defaultVolume=5
AudioFramework::maxUserVolume=5
BobController::bobGroupId=xxx
BobController::startTSClient=/home/ts3music/ts3.sh
QueryConnection::host=xxxxxxx
QueryConnection::user=xxxxxx
QueryConnection::passwd=xxxxxxx
QueryConnection::suppressLoopback=True
HistoryManager::historyFile=/home/ts3music/test.txt
HistoryManager::fillDeletedIds=True
PluginManager::PluginPath=Plugins
PlaylistManager::playlistPath=Playlists
YoutubeFactory::apiKey=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
YoutubeFactory::youtubedlpath=/usr/local/bin/youtube-dl
MainBot::logFile=log_ts3audiobot
MainBot::adminGroupId=xx
```

**Comment by Splamy** [comment]
Can you give me the excerpt from the logfile ?

**Comment by robotrono** [comment]
@Splamy yes:

```
[18:55:53]   Info: [============ TS3AudioBot started =============]
[18:55:53]   Info: [=== Date:         Saturday, 06 August 2016 ===]
[18:55:53]   Info: [=== Time:                         18:55:53 ===]
[18:55:53]   Info: [==============================================]
[18:55:53]   Info: [============ Initializing Commands ===========]
[18:55:53]   Info: [============ Initializing Modules ============]
[18:55:53]   Info: [=========== Initializing Factories ===========]
[18:55:53]   Info: [=========== Registering callbacks ============]
[18:55:53]   Info: [================= Finalizing =================]
[18:55:54]   Info: [============== Connected & Done ==============]
[18:56:04]  Debug: MB Got message from Akira: !pm
[18:56:04]  Debug: SM User Akira created session with the bot
[18:56:04]  Debug: AdminCheck called!
[18:56:04]  Debug: QC GetClientServerGroups called
[18:56:08]  Debug: MB Got message from Akira: !play [URL]https://www.youtube.com/watch?v=Pw-0pbY9JeU[/URL]
[18:56:08]Warning: TH Web error: [403] Forbidden
[18:56:08]  Debug: YT Ruined!

```

In Teamspeak:

```
<17:56:05> "TS3AudioBot" rejoined chat.
<17:56:05> "Akira": !play https://www.youtube.com/watch?v=Pw-0pbY9JeU
<17:56:05> "TS3AudioBot": Could not load (/usr/bin/python: can't open file '/usr/local/bin/youtube-dl/youtube_dl/__main__.py': [Errno 20] Not a directory

```

**Comment by Splamy** [comment]
Oooh, now I see it, you got the binary. I'm working with the uncompiled github project.
I will make a patch today which will detect and find the binary, too.

**Comment by robotrono** [comment]
I'll wating for the patch  :+1:

**Comment by Splamy** [comment]
@robotrono Ok, patch is out, please try the new version.

**Comment by robotrono** [comment]
Done, but dont works, the bot join to my channel and after for a little seconds disconnects.

Log:

```
[20:58:58]   Info: [============ TS3AudioBot started =============]
[20:58:58]   Info: [=== Date:         Saturday, 06 August 2016 ===]
[20:58:58]   Info: [=== Time:                         20:58:58 ===]
[20:58:58]   Info: [==============================================]
[20:58:58]   Info: [============ Initializing Commands ===========]
[20:58:58]   Info: [============ Initializing Modules ============]
[20:58:58]   Info: [=========== Initializing Factories ===========]
[20:58:58]   Info: [=========== Registering callbacks ============]
[20:58:58]   Info: [================= Finalizing =================]
[20:58:59]   Info: [============== Connected & Done ==============]
[20:59:04]  Debug: MB Got message from Akira: !play [URL]https://www.youtube.com/watch?v=Pw-0pbY9JeU[/URL]
[20:59:04]  Debug: SM User Akira created session with the bot
[20:59:04]Warning: TH Web error: [403] Forbidden
[20:59:04]  Debug: YT Ruined!
[20:59:05]  Debug: YT Saved!
[20:59:05]  Debug: AF stop old
[20:59:05]  Debug: BC Enqueing: music stop
[20:59:05]  Debug: AF ar start: Youtube ID:Pw-0pbY9JeU
[20:59:05]  Debug: BC Enqueing: music start https://r8---sn-nx57ynel.googlevideo.com/videoplayback?sparams=clen%2Cdur%2Cei%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Crequiressl%2Csource%2Cupn%2Cexpire&source=youtube&nh=IgpwcjAyLnNlYTAzKgkxMjcuMC4wLjE&clen=38858376&initcwndbps=3588750&ei=2YemV43EDI3H-wPnxYGgDg&dur=225.266&sver=3&expire=1470553145&ipbits=0&mn=sn-nx57ynel&mm=31&pl=32&mime=video%2Fmp4&id=o-AMDlGagn19bSN60TFmGlI1PoCy-pOC3ed_geanOWrcGh&gir=yes&mt=1470531277&ms=au&requiressl=yes&ip=2607%3A5501%3A1000%3Aef%3A%3Ac6ea&mv=m&upn=JHx99_jYppU&fexp=9405972%2C9419451%2C9422596%2C9428398%2C9431012%2C9431900%2C9433096%2C9433221%2C9433946%2C9435526%2C9436982%2C9437066%2C9437552%2C9438327%2C9438662%2C9438805%2C9439580%2C9440549%2C9440795%2C9441142%2C9441650%2C9441844%2C9442161%2C9442170%2C9442424%2C9443275%2C9443419%2C9443479%2C9443789%2C9443792&lmt=1458193813343182&key=yt6&itag=136&keepalive=yes&signature=6D67CDEBCEC5EB659233C5F5BCA535BE7203B614.AACA6577E0E7BB639D204DD1DEE9BD6E7B744604&ratebypass=yes
[20:59:05]  Debug: BC Enqueing: music volume 0.03
[20:59:05]  Debug: AF set volume: 3
[20:59:05]  Debug: BC Ressource started
[20:59:05]  Debug: BC Enqueing: callback on
[20:59:05]  Debug: BC now we are waiting for the bob
[20:59:05]  Debug: BC Enqueing: music unpause
[20:59:05]  Debug: BC Enqueing: audio on
[20:59:05]  Debug: BC Enqueing: whisper channel add 6649
[20:59:07]  Debug: BC user entered with GrId 34,452
[20:59:07]  Debug: BC user with correct UID found
[20:59:07]  Debug: BC bob is now officially running
[20:59:07]  Debug: BC sending to bobC: music stop
[20:59:07]  Debug: BC sending to bobC: music start https://r8---sn-nx57ynel.googlevideo.com/videoplayback?sparams=clen%2Cdur%2Cei%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Crequiressl%2Csource%2Cupn%2Cexpire&source=youtube&nh=IgpwcjAyLnNlYTAzKgkxMjcuMC4wLjE&clen=38858376&initcwndbps=3588750&ei=2YemV43EDI3H-wPnxYGgDg&dur=225.266&sver=3&expire=1470553145&ipbits=0&mn=sn-nx57ynel&mm=31&pl=32&mime=video%2Fmp4&id=o-AMDlGagn19bSN60TFmGlI1PoCy-pOC3ed_geanOWrcGh&gir=yes&mt=1470531277&ms=au&requiressl=yes&ip=2607%3A5501%3A1000%3Aef%3A%3Ac6ea&mv=m&upn=JHx99_jYppU&fexp=9405972%2C9419451%2C9422596%2C9428398%2C9431012%2C9431900%2C9433096%2C9433221%2C9433946%2C9435526%2C9436982%2C9437066%2C9437552%2C9438327%2C9438662%2C9438805%2C9439580%2C9440549%2C9440795%2C9441142%2C9441650%2C9441844%2C9442161%2C9442170%2C9442424%2C9443275%2C9443419%2C9443479%2C9443789%2C9443792&lmt=1458193813343182&key=yt6&itag=136&keepalive=yes&signature=6D67CDEBCEC5EB659233C5F5BCA535BE7203B614.AACA6577E0E7BB639D204DD1DEE9BD6E7B744604&ratebypass=yes
[20:59:08]  Error: Critical program failure! Logs will follow.
[20:59:08]  Error: MSG: 1541: the command failed to execute: invalid parameter size (missing permission:-1)
STACK:  at TS3Query.WaitBlock.WaitForMessage () <0x41b09980 + 0x00083> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.SendInternal (System.String command, TS3Query.Parameter[] parameter, TS3Query.Option[] options, System.Type targetType) <0x41b089e0 + 0x0032b> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.Send (System.String command, TS3Query.Parameter[] parameter, TS3Query.Option[] options) <0x41b08990 + 0x0001f> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.Send (System.String command, TS3Query.Parameter[] parameter) <0x41b08950 + 0x0002f> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.SendMessage (MessageTarget target, Int32 id, System.String message) <0x41b7aeb0 + 0x0018b> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.SendMessage (System.String message, ClientData client) <0x41b7ae50 + 0x0003f> in <filename unknown>:0 
  at TS3AudioBot.QueryConnection.SendMessage (System.String message, ClientData client) <0x41b7ae00 + 0x0002f> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) TS3AudioBot.QueryConnection:SendMessage (string,TS3Query.Messages.ClientData)
  at TS3AudioBot.BobController.SendMessageRaw (System.String message) <0x41b7ac50 + 0x000d3> in <filename unknown>:0 
  at TS3AudioBot.BobController.SendQueue () <0x41b7aaf0 + 0x00093> in <filename unknown>:0 
  at TS3AudioBot.BobController.OnBobConnect (System.Object sender, ClientEnterView e) <0x41b7a240 + 0x0027b> in <filename unknown>:0 
  at TS3AudioBot.QueryConnection.ExtendedClientEnterView (System.Object sender, ClientEnterView eventArgs) <0x41b7a1f0 + 0x0003b> in <filename unknown>:0 
  at TS3Query.TS3QueryClient+<InvokeEvent>c__AnonStorey0.<>m__0 () <0x41b7a0b0 + 0x000d9> in <filename unknown>:0 
  at TS3Query.DoubleThreadEventDispatcher.EnterEventLoop () <0x41b12730 + 0x00067> in <filename unknown>:0 
  at TS3Query.TS3QueryClient.EnterEventLoop () <0x41b126c0 + 0x0002e> in <filename unknown>:0 
  at TS3AudioBot.QueryConnection.EnterEventLoop () <0x41b12690 + 0x0001b> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) TS3AudioBot.QueryConnection:EnterEventLoop ()
  at TS3AudioBot.MainBot.Run () <0x41b125b0 + 0x00047> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) TS3AudioBot.MainBot:Run ()
  at TS3AudioBot.MainBot.Main (System.String[] args) <0x41a83d60 + 0x001bb> in <filename unknown>:0 
[20:59:08]   Info: Closing Mediaplayer...
[20:59:08]  Debug: AF stop old
[20:59:08]  Debug: BC sending to bobC: music stop
[20:59:08]  Debug: BC sending to bobC: audio off
[20:59:08]  Debug: BC initialted Bob-stop (isRunning=True)
[20:59:08]  Debug: BC start timeout
[20:59:08]   Info: BC Exiting bob
[20:59:08]  Debug: BC sending to bobC: exit
[20:59:08]   Info: Closing QueryConnection...
```

In teamspeak:

```
<19:59:00> "Akira": !play https://www.youtube.com/watch?v=Pw-0pbY9JeU
<19:59:31> Chat partner disconnected out of view.
```

**Comment by Splamy** [comment]
Uh, yeah, we had this problem recently too, it happens because the request/link is longer than 1024 chars, and teamspeak doesnt allow longer messages, so the pm to the bob failes.
Unfortunately we currently have no fix for that, the only thing you can do is restart the bot and try again.
I'm working on a workaround for that and/or a better handling of the error, but this will take a while.

**Comment by Splamy** [comment]
I'll close this issue, since the playlist problem is solved, the new problem is moved to #32
